### PR TITLE
optional server response analyse functions

### PR DESCRIPTION
**Pull request by doktordirk**
this should make it so easy to customize getting tokens and exp, that all types of backends should be supported

### CODE REVIEW

**Comment by doktordirk** [comment]
closes https://github.com/SpoonX/aurelia-authentication/issues/231

File: doc/baseConfig.md [hunk 1]
```diff
@@ -106,6 +108,15 @@ platform = 'browser';
 storage = 'localStorage';
 // The key used for storing the authentication response locally
 storageKey = 'aurelia_authentication';
+// optional function to extract the expiration date. takes the server responseas parameter
```
Comment (ricbermo) [inline doc/baseConfig.md]: there's a little typo. 

> takes the server responseas parameter

**Comment by RWOverdijk** [comment]
Must be rebased

File: doc/Quick start.md [hunk 2]
```diff
@@ -115,7 +115,14 @@ export class Login {
     }
 
     login() {
-        return this.authService.login(this.email, this.password)
+        return this.authService.login({email: this.email, password: this.password})
+        // // or for eg OpenIddict
```
Comment (RWOverdijk) [inline doc/Quick start.md]: Maybe put this elsewhere. A quick-start is an opinionated guide to get someone into the plugin quickly. Anything specific belongs on a different page.

Comment (doktordirk) [inline doc/Quick start.md]: true, i was just lazy again

**Comment by RWOverdijk** [comment]
Rename last two commit messages. One has a quote, the other is missing a space.

File: doc/api_authService.md [hunk 3]
```diff
@@ -375,19 +375,28 @@ Promise: response
 #### Examples
 
 ```js
-this.authService.signup('janedoe@example.com', 'securePasword')
```
Comment (RWOverdijk) [inline doc/api_authService.md:378]: Where is the signup example?

**Comment by RWOverdijk** [comment]
Feedback, but after 👍

File: doc/api_authService.md (hunk 3 above)
Comment (doktordirk) [inline doc/api_authService.md:378]: under signup :smile: 

the login example used wrongly signup instead of login

**Pull request merged**